ddtrace/_trace/sampling_rule.py

```diff
@@ -153,26 +153,25 @@ def check_tags(self, meta, metrics):
         tag_match = False
         for tag_key in self._tag_value_matchers.keys():
             value = meta.get(tag_key)
-            tag_match = self._tag_value_matchers[tag_key].match(str(value))
-            # If the value doesn't match in meta, check the metrics
-            if tag_match is False:
+            # it's because we're not checking metrics first before continuing
+            if value is None:
                 value = metrics.get(tag_key)
+                if value is None:
```

Comment:
I think this introduces a bug. This PR introduces an edge case where the value of some expected tags can be None.

Given `DatadogSampler(rules=[SamplingRule(sample_rate=0, tags={"key": "val", "key2": "val2"})])`)`, 
if a Span._meta contains `key2:val2` but not `key:val`. The span will match the sampling rule, even though it doesn't have the full list of expected tags.

A potential fix could be to always match on the value but store None as an empty string instead of `"None"`. 

cc: @ZStriker19